compiler/src/iree/compiler/Codegen/LLVMCPU/test/verify_vector_size_legality.mlir

```diff
@@ -0,0 +1,46 @@
+// RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-llvmcpu-verify-vector-size-legality))" --split-input-file %s --verify-diagnostics
+
+// expected-error @+1 {{One or more operations with large vector sizes (4096 bytes) were found:}}
+func.func @large_vector_without_native_vector_size(%arg0 : vector<16x64xf32>) -> vector<16x64xf32> {
+  // expected-note-re @+1 {{math.exp}}
+  %0 = math.exp %arg0 : vector<16x64xf32>
+  // expected-note-re @+1 {{return}}
+  return %0 : vector<16x64xf32>
+}
+
+// -----
+
+// expected-error @+1 {{One or more operations with large vector sizes (32768 bytes) were found:}}
+func.func @large_vector_with_native_vector_size(%arg0 : vector<16x64x64xf32>) -> vector<16x64x64xf32> attributes {
+  hal.executable.target = #hal.executable.target<"llvm-cpu", "xyz", {native_vector_size = 64}>
+} {
+  // expected-note-re @+1 {{math.exp}}
+  %0 = math.exp %arg0 : vector<16x64x64xf32>
+  // expected-note-re @+1 {{return}}
+  return %0 : vector<16x64x64xf32>
+}
+
+// -----
+
+// expected-error @+1 {{One or more operations with large vector sizes (32768 bytes) were found:}}
+func.func @large_contract_with_native_vector_size(%lhs : vector<32x64xf32>, %rhs : vector<32x64xf32>, %acc : vector<32x32xf32>) -> vector<32x32xf32> attributes {
+  hal.executable.target = #hal.executable.target<"llvm-cpu", "xyz", {native_vector_size = 64}>
+} {
+  // expected-note-re @+1 {{vector.contract}}
+  %0 = vector.contract {
+      indexing_maps = [affine_map<(d0, d1, d2) -> (d0, d2)>,
+                       affine_map<(d0, d1, d2) -> (d1, d2)>,
+                       affine_map<(d0, d1, d2) -> (d0, d1)>],
+      iterator_types = ["parallel", "parallel", "reduction"],
+      kind = #vector.kind<add>}
+    %lhs, %rhs, %acc : vector<32x64xf32>, vector<32x64xf32> into vector<32x32xf32>
+  return %0 : vector<32x32xf32>
+}
+
+// -----
+
+// CHECK: func.func @scalable_vector
+func.func @scalable_vector(%arg0 : vector<[1]x64xf32>) -> vector<[1]x64xf32> {
+  %0 = math.exp %arg0 : vector<[1]x64xf32>
+  return %0 : vector<[1]x64xf32>
+}
```

Comment:
Would you mind adding a few more tests for scalable vectors? 

In general, for the purpose of this pass, I think that it's safe to ignore scalability and just make sure that the logic still works in the presence of scalable dims.